src/Sql/Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs

```diff
@@ -441,22 +582,71 @@ protected override AzureSqlManagedDatabaseModel GetEntity()
 
             switch (ParameterSetName)
             {
+                case PointInTimeDeletedCrossSubscriptionRestoreFromNameAndResourceGroupSet:
+                    model.CrossSubscriptionTargetManagedInstanceId = "/subscriptions/" + TargetSubscriptionId +
+                        "/resourceGroups/" + TargetResourceGroupName +
+                        "/providers/Microsoft.Sql/managedInstances/" + TargetInstanceName;
+                    model.CrossSubscriptionRestorableDroppedDatabaseId = "/subscriptions/" + SubscriptionId +
+                        "/resourceGroups/" + ResourceGroupName +
+                        "/providers/Microsoft.Sql/managedInstances/" + InstanceName +
+                        "/restorableDroppedDatabases/" + Name + "," + DeletionDate.ToFileTimeUtc();
+                    break;
+                case PointInTimeCrossSubscriptionRestoreFromNameAndResourceGroupSet:
+                    model.CrossSubscriptionTargetManagedInstanceId = "/subscriptions/" + TargetSubscriptionId +
+                            "/resourceGroups/" + TargetResourceGroupName +
+                            "/providers/Microsoft.Sql/managedInstances/" + TargetInstanceName;
+                    model.CrossSubscriptionSourceDatabaseId = "/subscriptions/" + SubscriptionId +
+                        "/resourceGroups/" + ResourceGroupName +
+                        "/providers/Microsoft.Sql/managedInstances/" + InstanceName +
+                        "/databases/" + Name;
+                    break;
+                case PointInTimeCrossSubscriptionRestoreFromInputObjectParameterSet:
+                    SubscriptionId = new ResourceIdentifier(InputObject.Id).Subscription;
+                    ResourceGroupName = InputObject.ResourceGroupName;
+                    InstanceName = InputObject.ManagedInstanceName;
+                    if (InputObject is AzureSqlDeletedManagedDatabaseBackupModel)
+                    {
+                        model.CrossSubscriptionRestorableDroppedDatabaseId = InputObject.Id;
+                    }
+                    else
+                    {
+                        model.CrossSubscriptionSourceDatabaseId = InputObject.Id;
+                    }
+                    model.CrossSubscriptionTargetManagedInstanceId = "/subscriptions/" + TargetSubscriptionId +
+                            "/resourceGroups/" + TargetResourceGroupName +
+                            "/providers/Microsoft.Sql/managedInstances/" + TargetInstanceName;
+                    break;
+                case PointInTimeCrossSubscriptionRestoreFromResourceIdParameterSet:
+                    var sourceResourceInfo = new ResourceIdentifier(ResourceId);
+                    SubscriptionId = sourceResourceInfo.Subscription;
+                    ResourceGroupName = sourceResourceInfo.ResourceGroupName;
+                    InstanceName = sourceResourceInfo.ParentResource.Split(new[] { '/' })[1];
```

Comment:
```suggestion
                    InstanceName = sourceResourceInfo.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
```
nit

Reply:
I don't think instance name would be `last` in this case, but `last -1`. SourceResourceInfo is info about database where type is `Microsoft.Sql/managedInstances/databases` or `Microsoft.Sql/managedInstances/restorableDroppedDatabases` if we are restoring deleted database.

Any suggestion in this case?